core/modules/taxonomy/taxonomy.module

```diff
@@ -1852,6 +1853,53 @@ function taxonomy_field_widget_form(&$form, &$form_state, $field, $instance, $la
   return $element;
 }
 
+/**
+ * Implements hook_field_widget_form_alter().
+ */
+function taxonomy_field_widget_form_alter(&$element, &$form_state, $context) {
```

Comment:
@bugfolder, this is what I think the function should look like (which is, I think, what you had earlier before I confused things with my erroneous comments):

```php
/**
 * Implements hook_field_widget_form_alter().
 */
function taxonomy_field_widget_form_alter(&$element, &$form_state, $context) {
  $field = $context['field'];
  $widget_type = $context['instance']['widget']['type'];


  if ($field['type'] == 'taxonomy_term_reference' && in_array($widget_type, array('options_buttons', 'options_select'))) {
    $trees = $field['settings']['allowed_values'];
    
    foreach ($trees as $tree) {
      if ($vocabulary = taxonomy_vocabulary_load($tree['vocabulary'])) {
        if ($terms = taxonomy_get_tree($vocabulary->machine_name, $tree['parent'])) {
          foreach ($terms as $term) {
            if (isset($element['#options'][$term->tid])) {
              $element[$term->tid] = array('#indentation' => $term->depth);
            }
          }
        }
      }
    }
  }
}
```